From 37b9966e2525790843ab302a5b8009853a7905a0 Mon Sep 17 00:00:00 2001
From: dann frazier <dann.frazier@canonical.com>
Date: Mon, 27 Jan 2020 18:45:17 -0700
Subject: [PATCH] test: Synchronize journal before reading from it

There's a race condition in the sysuser test where it may try to read
entries from the journal before they are available. Fix it by adding a
`journalctl --sync` call.

Origin: https://github.com/systemd/systemd/commit/37b9966e2525790843ab302a5b8009853a7905a0
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1776654
Last-Update: 2020-01-28
Applied-Upstream: https://github.com/systemd/systemd/commit/37b9966e2525790843ab302a5b8009853a7905a0

---
 test/TEST-21-SYSUSERS/test.sh | 1 +
 1 file changed, 1 insertion(+)

Index: systemd/test/TEST-21-SYSUSERS/test.sh
===================================================================
--- systemd.orig/test/TEST-21-SYSUSERS/test.sh
+++ systemd/test/TEST-21-SYSUSERS/test.sh
@@ -114,6 +114,7 @@ test_run() {
         prepare_testdir ${f%.input}
         cp $f $TESTDIR/usr/lib/sysusers.d/test.conf
         systemd-sysusers --root=$TESTDIR 2> /dev/null
+        journalctl --sync
         journalctl -t systemd-sysusers -o cat | tail -n1 > $TESTDIR/tmp/err
         if ! diff -u $TESTDIR/tmp/err  ${f%.*}.expected-err; then
             echo "**** Unexpected error output for $f"
